src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -143,4 +143,21 @@ public Book updateBook(@RequestBody Book book, @PathVariable Long id) {
 		return bookRepository.findByIsbn(isbn).map(book -> new ResponseEntity<>(book, HttpStatus.OK))
 				.orElseGet(() -> new ResponseEntity<>(openLibraryDelegate.findBookByIsbn(isbn), HttpStatus.CREATED));
 	}
+
+	/**
+	 * This method is to find a book by publisher and genre and year
+	 *
+	 * @param publisher: publisher of the book (Object)
+	 * @param genre:     genre of the book (Object)
+	 * @param year:      year of the book (Object)
+	 *
+	 * @return {@link Book}
+	 */
+
+	@GetMapping("/parameters")
+	public List<Book> findByPublisherAndGenreAndYear(@RequestParam(required = false) String publisher,
```

Comment:
Nice! The Trello card does not ask you to provide and endpoint but is nice that you have added it!